pkg/storage/snapshot/snapshot.go

```diff
@@ -850,12 +849,28 @@ func (ctrl *VMSnapshotController) getVolumeSnapshotStatus(vm *kubevirtv1.Virtual
 		}
 	}
 
-	sc, err := ctrl.getVolumeStorageClass(vm.Namespace, volume)
+	pvc, err := ctrl.getPVCFromVolume(vm.Namespace, volume)
 	if err != nil {
 		return kubevirtv1.VolumeSnapshotStatus{Name: volume.Name, Enabled: false, Reason: err.Error()}
 	}
 
-	snap, err := ctrl.getVolumeSnapshotClass(sc)
+	if !isPVCBound(pvc) {
+		return kubevirtv1.VolumeSnapshotStatus{
+			Name:    volume.Name,
+			Enabled: false,
```

Comment:
I'm not necessarily against this change as-is but what do you think about keeping `Enabled` computation as-is but adding a `Bound` flag to `VolumeSnapshotStatus`.  Then:

Clone controller could then check that `Enabled` && `Bound`?

OR 

Clone controller just checks `Enabled` and Snapshot controller waits for PVCs to be bound (if enabled is true)?  Once all PVCs are bound we consider that the start of the snapshot operation and initialize the deadline.  I think this scheme will address #12082

Reply:
@mhenriks but what do you do with volumes in a stopped VM that are not bound? they are currently marked as snapshot enabled, though currently they are skipped. waiting for the to be bound will reach timeout.. they are not actually snapshotable at this state..
And adding `bound` to volumesnapshotstatus doesnt seem right to me.. I think having in the reason pvc not bound as to why snapshot is disabled is better..

Reply:
> but what do you do with volumes in a stopped VM that are not bound? they are currently marked as snapshot enabled, though currently they are skipped. waiting for the to be bound will reach timeout.. they are not actually snapshotable at this state..

Was thinking we would wait until all PVCs are bound then start deadline and do the real work (freeze/create VolumeSnapshots/etc).  This seems to be more in line with eventual consistency goal than skipping, no?

> And adding bound to volumesnapshotstatus doesnt seem right to me.. I think having in the reason pvc not bound as to why snapshot is disabled is better..

I'm not sure I like it either, but given eventual consistency goal, I think it is more important to report to the user what is "theoretically possible" (current definition of enabled).  But what is "possible now" may also be important.

I feel that clone api should only care about what is theoretically possible and push all logic down to snapshot layer

Reply:
> This seems to be more in line with eventual consistency goal

In the end of the day I think both eventual consistency and failing fast approaches have their merits. Failing fast is often easier to reason about with a quick feedback look, while on the other hand eventual consistency is very helpful when multiple events occur nearly at the same time and we want ordering to not matter.

Maybe we should try to think what's the added benefit of the eventual consistency approach in this specific case. For example, is it expected that users would create a snapshot + start a VM approximately at the same time?

Reply:
> And adding bound to volumesnapshotstatus doesnt seem right to me.. I think having in the reason pvc not bound as to why snapshot is disabled is better..

That's fine, but keep in mind that from a clone's perspective it matters if the volume is unsnapshotable in essence or just not bound yet because of the dv template case. An alternative is that the clone can rely on the `Reason` section to figure it out.

Reply:
> In the end of the day I think both eventual consistency and failing fast approaches have their merits.

I think "failing" may be better than partial or incomplete snapshots.  But what is there for the user (or clone controller) to do when a snapshot fails?  If the answer is "keep retrying" then I think it would be better for the snapshot controller to wait for VM pvcs to be bound

Reply:
Hey @mhenriks and @ShellyKa13!

I think we need to reach a decision here :) https://github.com/kubevirt/kubevirt/pull/12082 is also blocked due to this discussion.

> But what is there for the user (or clone controller) to do when a snapshot fails? if the answer is "keep retrying" then I think it would be better for the snapshot controller to wait for VM pvcs to be bound

Eventually, the user (in both the cloning and snapshotting case) would need to start the VM and stop it again.

I think this discussion can be boiled down to:
* What's more **visible** to the user? A **failed** snapshot/clone, or a snapshot/clone that is waiting with an **event** or something similar?
* Do we suspect a use-case in which VMs and snapshots/clones would be created at the same time (e.g. with gitops), therefore eventual consistency is crucial here?

Reply:
> (not forever by default)

Sounds like a good compromise to me - waiting for the PVC to be bound for a relatively short amount of time.

This way we gain the benefits of eventual consistency, while still failing relatively fast.
WDYT @ShellyKa13?

Reply:
Yes, Ill add that we should also move the check of the volumes for snapshot to before "locking" the vm so in case a vm has never started it gives the user the option to start is so the volumes will get bound.
The time it will wait is the already existing deadline timeout which is by default 5 min and can be configured per snapshot.

Reply:
Thank you for the conversation! :)
I'll use the same approach in https://github.com/kubevirt/kubevirt/pull/12082.